spark/src/main/scala/org/apache/spark/sql/delta/hooks/AutoCompact.scala

```diff
@@ -0,0 +1,257 @@
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.delta.hooks
+
+import org.apache.spark.sql.delta.skipping.clustering.ClusteredTableUtils
+import org.apache.spark.sql.delta._
+import org.apache.spark.sql.delta.actions._
+import org.apache.spark.sql.delta.commands.{DeltaOptimizeContext, OptimizeExecutor}
+import org.apache.spark.sql.delta.commands.optimize._
+import org.apache.spark.sql.delta.metering.DeltaLogging
+import org.apache.spark.sql.delta.sources.DeltaSQLConf
+import org.apache.spark.sql.delta.stats.AutoCompactPartitionStats
+
+import org.apache.spark.sql.SparkSession
+import org.apache.spark.sql.catalyst.catalog.CatalogTable
+import org.apache.spark.sql.catalyst.expressions.Expression
+import org.apache.spark.sql.internal.SQLConf
+
+/**
+ * A trait for post commit hook which compacts files in a Delta table. This hook acts as a cheaper
+ * version of the OPTIMIZE command, by attempting to compact files to a smaller size.
+ *
+ * Auto Compact chooses files to compact greedily by looking at partition directories which
+ * have the largest number of files that are under a certain size threshold and launches a bounded
+ * number of optimize tasks based on the capacity of the cluster.
+ */
+trait AutoCompactBase extends PostCommitHook with DeltaLogging {
+
+  override val name: String = "Auto Compact"
+
+  private[delta] val OP_TYPE = "delta.commit.hooks.autoOptimize"
+
+  /**
+   * This method returns the type of Auto Compaction to use on a delta table or returns None
+   * if Auto Compaction is disabled.
+   * Prioritization:
+   *   1. The highest priority is given to [[DeltaSQLConf.DELTA_AUTO_COMPACT_ENABLED]] config.
+   *   2. Then we check if the deprecated property `DeltaConfigs.AUTO_OPTIMIZE` is set. If yes, then
+   *      we return [[AutoCompactType.Legacy]] type.
+   *   3. Then we check the table property [[DeltaConfigs.AUTO_COMPACT]].
+   *   4. If none of 1/2/3 are set explicitly, then we return None
+   */
+  def getAutoCompactType(conf: SQLConf, metadata: Metadata): Option[AutoCompactType] = {
+    // If user-facing conf is set to something, use that value.
+    val autoCompactTypeFromConf =
+      conf.getConf(DeltaSQLConf.DELTA_AUTO_COMPACT_ENABLED).map(AutoCompactType(_))
+    if (autoCompactTypeFromConf.nonEmpty) return autoCompactTypeFromConf.get
+
+    // If user-facing conf is not set, use what table property says.
+    val deprecatedFlag = DeltaConfigs.AUTO_OPTIMIZE.fromMetaData(metadata)
+    val autoCompactTypeFromPropertyOrDefaultValue = deprecatedFlag match {
+      case Some(true) =>
+        Some(AutoCompactType.Enabled)
```

Comment:
For both, auto_optimize and auto_compact we are returning `Enabled`.

Reply:
If we are not splitting Enabled in two (Enabled, Legacy), this comment is not valid anymore